### ISSUE TITLE
Nuitka Release 0.6.4 doesn't add icon to windows 10 EXE file with MinGW64

### ISSUE THREAD

**Issue by shoemak** [Nuitka/Nuitka#431]
Hello!
After updating to new Nuitka Release 0.6.4 on Windows 10 my standalone executable no longer attach .ico picture by --windows-icon=c:\icnfolder\icn.ico command.
There was nothing changed in compilation script - on previous version of Nuitka all worked.

Please check if this is a real bug.
Thank you for your work!

**Comment by kayhayen** [comment]
Can you send the full output with `--show-scons` ?

**Comment by shoemak** [comment]
> Can you send the full output with `--show-scons` ?
Yes, I can, but it is possible only tomorrow.
Now my compillation script looks like:
>python -m nuitka --standalone --follow-imports --mingw64 --plugin-enable=qt-plugins --remove-output --assume-yes-for-downloads --show-progress --windows-disable-console --windows-icon=c:\icnfolder\icn.ico c:\test\test.py

I will remove "--show-progress" and will insert "--show-scons"

**Comment by kayhayen** [comment]
Looks good to me @shoemak

**Comment by shoemak** [comment]
@kayhayen Hello! 
The full output with --show-scons is quite long... I've attched file containing the output.
Is this the data you need?
[scons.txt](https://github.com/Nuitka/Nuitka/files/3289396/scons.txt)

**Comment by kayhayen** [comment]
So I am indeed seeing the call to windres and I think it is a regression related to a MinGW64 workaround that I have done for scalability and Scons for long command lines.

Can you remove this code from `SingleExe.scons` in your Nuitka install:

```
# Hack to make Scons use tempfile for gcc linking.
if gcc_mode:
    tmp_linker_filename = os.path.join(source_dir, "@sources.tmp")

    env["LINKCOM"] = env["LINKCOM"].replace(
        "$SOURCES", "@%s" % env.get("ESCAPE", lambda x: x)(tmp_linker_filename)
    )

    with open(tmp_linker_filename, "w") as tmpfile:
        for filename in source_files:
            filename = ".".join(filename.split(".")[:-1]) + ".o"

            if os.name == "nt":
                filename = filename.replace(os.path.sep, "/")

            tmpfile.write('"%s"\n' % filename)

        tmpfile.write(env.subst("$SOURCES"))

```

I think it may fail to preserve the "windres" object file, causing the issue right now.

Yours,
Kay

**Comment by shoemak** [comment]
@kayhayen Hello!
I have removed that code from SingleExe.scons and recompiled my project.
Icon in .exe file has appeared!
Thank you for your work and time!

**Comment by kayhayen** [comment]
Ok, the real solution is probably to move the icon handling out of there entirely, and I will focus on that for 0.6.5 then. For now, this is a workaround that most people can employ. Obviously using MSVC would also work, but it's nothing I like to recommend.

**Comment by kayhayen** [comment]
I changed my mind, reading icon files will be a while off, so I made a fix for this issue, intended to be in 0.6.5, currently it's on the factory branch:

https://nuitka.net/doc/factory.html

Yours,
Kay

**Comment by kayhayen** [comment]
Released as 0.6.5
